src/plugins/data/server/index_patterns/index_patterns_service.ts

```diff
@@ -53,43 +53,55 @@ export interface IndexPatternsServiceStartDeps {
   logger: Logger;
 }
 
-export class IndexPatternsService implements Plugin<void, IndexPatternsServiceStart> {
+export class IndexPatternsServiceProvider implements Plugin<void, IndexPatternsServiceStart> {
+  private uiSettings?: CoreStart['uiSettings'];
+  private fieldFormats?: FieldFormatsStart;
+  private logger?: Logger;
+
   public setup(
     core: CoreSetup<DataPluginStartDependencies, DataPluginStart>,
     { expressions }: IndexPatternsServiceSetupDeps
   ) {
     core.savedObjects.registerType(indexPatternSavedObjectType);
     core.capabilities.registerProvider(capabilitiesProvider);
 
-    registerRoutes(core.http);
+    registerRoutes(core.http, this);
 
     expressions.registerFunction(getIndexPatternLoad({ getStartServices: core.getStartServices }));
   }
 
   public start(core: CoreStart, { fieldFormats, logger }: IndexPatternsServiceStartDeps) {
-    const { uiSettings } = core;
+    this.uiSettings = core.uiSettings;
```

Comment:
I think we agreed that these changes were unnecessary

Reply:
No, you wanted to Zoom about this.

Reply:
Agreed over Zoom to use `core.getStartServices` for this.